### PR TITLE
Feature/config/root

### PR DESCRIPTION
**Pull request by J0Santos**
Implemented root_handler method now there's a root attribute in parser and when my handler receives a valid set of commands the root variable is updated.
Tests are passing

Made some overall changes to maria's code everything's good
This branch is ready to merge with main

### CODE REVIEW

File: srcs/Parser.cpp [hunk 1]
```diff
@@ -19,44 +19,51 @@ unsigned int	Parser::get_listen(void) const { return (this->listen); }
...
 
 	if (command.size() < 2 ) {
-		this->set_error(new Error());
+		this->error = new Error();
 		return (false);
 	}
 
 	s << command[1];
 	s >> port;
 	if (port <= 0) {
```
Comment (Dutesier) [inline srcs/Parser.cpp:48]: Port should also be lower than 65535

File: srcs/Parser.cpp [hunk 2]
```diff
@@ -19,44 +19,51 @@ unsigned int	Parser::get_listen(void) const { return (this->listen); }
...
+	this->listen = port;
+	return (true);
+}
+
+// This function returns true if the command is valid so that it can be
+// successfully stored in this->root. In case of error, the function returns
+// false and this->error stops being nullptr to an instance of the Error class.
+bool	Parser::root_handler(std::vector<std::string> command) {
+
+	std::string root;
```
Comment (Dutesier) [inline srcs/Parser.cpp:62]: Is std::string root ever used here? (line 62)

File: tests/test_parser.cpp [hunk 3]
```diff
@@ -9,54 +9,86 @@
...
+	EXPECT_TRUE(this->parser->listen_handler(command));
+	EXPECT_EQ(this->parser->get_error(), nullptr);
+	EXPECT_EQ(this->parser->get_listen(), 443);
+
+	// Testing default value
+	this->Reset();
+	EXPECT_EQ(this->parser->get_listen(), 80);
+}
+
+
```
Comment (Dutesier) [inline tests/test_parser.cpp:68]: Might be a better practice to separate tests. This way we can run single tests and better debug what test is failing.

File: srcs/Parser.cpp (hunk 1 above)
Comment (Dutesier) [inline srcs/Parser.cpp:48]: Shouldn't port also be lower than 65535?

File: srcs/Parser.cpp (hunk 2 above)
Comment (J0Santos) [reply srcs/Parser.cpp:62]: nop, mb. Can be removed

**Review by Dutesier — APPROVED** [review]

**Pull request merged**